File: C++/fuelV2.cpp
/*
 *  Programmer:     Corey Jenkins
 *  Date:           March 24, 2024
 *  Filename:       fuelV2.cpp
 *  Purpose:        Using a combination of a switch and while loops for input validation to select
 *                  which type of fuel and how many gallons. It will then calculate the price by the amount.
 *                  of gallons inputted.
*/

#include <iostream>
#include <string>
using namespace std;

int main()
{
    string regular = "Regular";
    string plus = "Plus";
    string supreme = "Supreme";

    float regularPrice = 3.99;
    float plusPrice = 4.99;
    float supremePrice = 5.99;

    int gallons;
    float total;

    int fuelSelect;
    cout << "Select fuel: " << endl;
    cout << "1 - Regular" << endl;
    cout << "2 - Plus" << endl; 
    cout << "3 - Supreme" << endl;
    cin >> fuelSelect;

    while (fuelSelect > 3 || fuelSelect <= 0)
    {
        cout << "Error! Please select a fuel grade from below. These are your options:" <<endl;
        cout << "1 - Regular" << endl;
        cout << "2 - Plus" << endl;
        cout << "3 - Supreme" << endl;
        cin >> fuelSelect;
    }

    switch (fuelSelect)
    {
        case 1:
        
            cout << "You have selected: " << regular << endl;
            cout << "Please enter the amount of gallons: " << endl;
            cin >> gallons;
            while(gallons <= 0)
            {
                cout << "Error! Please enter a postive number. Please enter the amount of gallons again:" << endl;
                cin >> gallons;
            }

            total = gallons * regularPrice;
            cout << "Thank you. Here's your reciept: " << total << endl;
            break;

        case 2:

            cout << "You have selected: " << plus << endl;
            cout << "Please enter the ammount of gallons:" << endl;
            cin >> gallons;
            while (gallons <= 0)
            {
                cout << "Error! Please enter a postive number. Please enter the amount of gallons again:" << endl;
                cin >> gallons;
            }

            total = gallons * plusPrice;
            cout << "Thank you. Here is your receipt: " <<  total << "." << endl;
            break;

        case 3:
            cout << "You have selected: " << supreme << endl;
            cout << "Please enter the amount of gallons: " << endl;
            cin >> gallons;
            while (gallons <=0)
            {
                cout << "Error! Please enter a postive number. Please enter the amount of gallons again: " << endl;
                cin >> gallons;
            }

            total = gallons * supremePrice;
            cout << "Thank you. Here is your receipt: " << total << endl;
    }
    // End Module Main
    return 0; 
}

File: Java/FuelV2.java
/*
 *  Programmer:     Corey Jenkins
 *  Date:           March 24, 2024
 *  Filename:       fuelV2.java
 *  Purpose:        Using a combination of a switch and while loops for input validation to select
 *                  which type of fuel and how many gallons. It will then calculate the price by the amount.
 *                  of gallons inputted.
*/

import java.util.Scanner;

public class FuelV2 {
    public static void main(String[] args) {
        Scanner fuel = new Scanner(System.in);
        String regular = "Regular";
        String plus = "Plus";
        String supreme = "Supreme";

        double regularPrice = 3.99;
        double plusPrice = 4.99;
        double supremePrice = 5.99;

        int gallons;
        double total;

        int fuelSelect;

        System.out.println("Please select fuel type: ");
        System.out.println("1 - Regular");
        System.out.println("2 - Plus");
        System.out.println("3 - Supreme");
        fuelSelect = fuel.nextInt();
        while (fuelSelect <= 0 || fuelSelect > 3) {
            System.out.println("Error! Please select fuel type below. These are your options:");
            System.out.println("1 - Regular");
            System.out.println("2 - Plus");
            System.out.println("3 - Supreme");
            fuelSelect = fuel.nextInt();
        }

        switch (fuelSelect) {
            case 1:
                System.out.println("You have selected: " + regular + ".");
                System.out.println("Please enter gallons");
                gallons = fuel.nextDouble();
                while (gallons <= 0) {
                    System.out.println(
                            "Error! Please enter a positive number. Please enter the amount of gallons again.");
                    gallons = fuel.nextDouble();
                }

                total = gallons * regularPrice;
                System.out.println("Thank you! Here's your reciept: " + total);
                break;

            case 2:
                System.out.println("You have selected: " + plus + ".");
                System.out.println("Please enter gallons");
                gallons = fuel.nextDouble();
                while (gallons <= 0) {
                    System.out.println(
                            "Error! Please enter a positive number. Please enter the amount of gallons again.");
                    gallons = fuel.nextDouble();
                }

                total = gallons * plusPrice;
                System.out.println("Thank you! Here's your reciept: " + total);
                break;

            case 3:
                System.out.println("You have selected: " + supreme + ".");
                System.out.println("Please enter gallons");
                gallons = fuel.nextDouble();
                while (gallons <= 0) {
                    System.out.println(
                            "Error! Please enter a positive number. Please enter the amount of gallons again.");
                    gallons = fuel.nextDouble();
                }

                total = gallons * supremePrice;
                System.out.println("Thank you! Here's your reciept: " + total);
                break;
        }
    }
}